test/functional/tests/io_class/test_io_class_eviction_priority.py

```diff
@@ -0,0 +1,139 @@
+
+    with TestRun.step("Resetting cache stats"):
+        cache.purge_cache()
+        cache.reset_counters()
+
+    with TestRun.step("Checking initial occupancy"):
+        for io_class in io_classes:
+            occupancy = get_io_class_occupancy(cache, io_class.id)
+            if occupancy.get_value() != 0:
+                TestRun.LOGGER.error(
+                    f"Incorrect inital occupancy for ioclass id: {io_class.id}."
+                    f" Expected 0, got: {occupancy}"
+                )
+
+    with TestRun.step(f"To A, B and C directories perform IO with size of max io_class occupancy"):
+        for io_class in io_classes[0:3]:
+            run_io_dir(
+                f"{io_class.dir_path}/tmp_file",
+                int((io_class.max_occupancy * cache_size) / Unit.Blocks4096),
+            )
+
+    with TestRun.step("Check if each ioclass reached it's occupancy limit"):
+        for io_class in io_classes[0:3]:
+            actuall_occupancy = get_io_class_occupancy(cache, io_class.id)
+
+            occupancy_limit = (
+                (io_class.max_occupancy * cache_size)
+                .align_down(Unit.Blocks4096.get_value())
+                .set_unit(Unit.Blocks4096)
+            )
+
+            if not isclose(actuall_occupancy.value, occupancy_limit.value, rel_tol=0.1):
+                TestRun.LOGGER.error(
+                    f"Occupancy for ioclass {io_class.id} does not match. "
+                    f"Limit: {occupancy_limit}, actuall: {actuall_occupancy}"
+                )
+
+        if get_io_class_occupancy(cache, io_classes[3].id).value != 0:
+            TestRun.LOGGER.error(
+                f"Occupancy for ioclass {io_classes[3].id} should be 0. "
+                f"Actuall: {actuall_occupancy}"
+            )
+
+    with TestRun.step(
+        "Perform IO to the fourth directory and check "
+        "if other partitions are evicted in a good order"
+    ):
+        target_io_class = io_classes[3]
+        io_classes_to_evict = io_classes[:3][::-1]  # List is ordered by eviction priority
+        io_classes_evicted = []
+        for io_class in io_classes_to_evict:
+            run_io_dir(
+                f"{target_io_class.dir_path}/tmp_file",
+                int((io_class.max_occupancy * cache_size) / Unit.Blocks4096),
+            )
+            part_to_evict_end_occupancy = get_io_class_occupancy(cache, io_class.id, percent=True)
+
+            # Since number of evicted cachelines is always >= 128, occupancy is checked
+            # with approximation
+            if not isclose(part_to_evict_end_occupancy, 0, abs_tol=4):
+                TestRun.LOGGER.error(
+                    f"Wrong percent of cache lines evicted from part {io_class.id}. "
+                    f"Meant to be evicted {io_class.max_occupancy*100}%, actaully evicted "
+                    f"{io_class.max_occupancy*100-part_to_evict_end_occupancy}%"
```

Comment:
I think that there could be a check whether other partitions are not evicted

Reply:
Check added